apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/ServiceVersionUtil.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.configuration;
+
+import javax.annotation.Nullable;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+
+public class ServiceVersionUtil {
+
+    public static String getDefaultServiceVersion() {
+        return getDefaultServiceVersion(System.getProperty("sun.java.command"));
+    }
+
+    private static String getDefaultServiceVersion(@Nullable String command) {
```

Comment:
There's some duplication from ServiceNameUtil. What about merging the two and returning a single `ServiceInfo` Object:

```java
class ServiceInfo {
    private final String serviceName;
    private final String serviceVersion;
    private final boolean isContainer; // or Boolean with null = unknown?

    public static ServiceInfo create() {
        return create(System.getProperties());
    }
    public static ServiceInfo create(Properties properties) {
        // parsing sun.java.command
        // delegate to create(Manifest manifest) if command is a jar file
        // merge ServiceInfo if required
    }

    public static ServiceInfo create(Manifest manifest) {
        ...
    }

    ServiceInfo merge(ServiceInfo other) {
        // create new immutable copy and merge the two with one having precedence
    }
   ...
}
```

Reply:
This overlaps (in a nice way, actually) with the changes in https://github.com/elastic/apm-agent-java/pull/1726/files.

Maybe we don't need the isContainer flag and just not populate the `serviceVersion` instead.

Reply:
@felixbarny Would you please merge #1921? It makes creating `ServiceInfo` easier.

Reply:
Merged

Reply:
I merged ServiceNameUtil and ServiceVersionUtil. Can you please have a l look at my changes?

Reply:
Looks great, much cleaner!